src/components/jobDetailsComponent/jobDetails.tsx

```diff
@@ -43,7 +42,7 @@ const JobDetails: React.FC = () => {
   const renderButtons = () => {
     return job.userId === user.id ? <PosterButtons /> : <UserButtons />;
   };
-  const images = job.images.map(img => img.url);
+  const images = job.images ? job.images.map(img => img.url) : [];
```

Comment:
if job.images is an empty array it will be true too, so it's better to check tor length !== 0 or length > 0